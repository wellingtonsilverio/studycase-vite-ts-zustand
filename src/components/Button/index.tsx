interface Props {
  disabled?: boolean;
  children?: React.ReactNode;
  onClick?: () => void;
}

const Button: React.FC<Props> = ({ disabled, children, onClick }) => {
  return (
    <button onClick={disabled ? undefined : onClick} disabled={disabled}>
      {children}
    </button>
  );
};

export default Button;
