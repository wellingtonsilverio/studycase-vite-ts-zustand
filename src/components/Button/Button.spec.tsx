import { render, screen } from '@testing-library/react';
import Button from '.';

describe('Button Component', () => {
  it('should render button', () => {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot();
  });
  it('should render text inside component', () => {
    render(<Button>Hi!</Button>);

    expect(screen.getByText('Hi!')).toBeInTheDocument();
  });
});
