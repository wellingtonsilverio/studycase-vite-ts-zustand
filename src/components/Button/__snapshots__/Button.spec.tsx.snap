// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render button 1`] = `
<div>
  <button />
</div>
`;
